src/dataflow/src/decode/csv.rs

```diff
@@ -114,14 +116,13 @@ where
                                     csv_core::ReadRecordResult::Record => {
                                         if bounds_valid != n_cols {
                                             events_error += 1;
-                                            error!(
-                                                "CSV error: expected {} columns, got {}. Ignoring row.",
-                                                n_cols, bounds_valid,
-                                            );
+                                            session.give((Err(DataflowError::DecodeError(
+                                                DecodeError::Text(format!("CSV error: expected {} columns, got {}. Ignoring row.", n_cols, bounds_valid))
```

Comment:
"Ignoring row" might be confusing to the user because it suggests that there is some way to get the other rows with this one filtered out. Which right now... there isn't. So even though it's true from an internal system perspective, we should probably avoid this sort of language from a user perspective.

Reply:
I will update the language on this! Perhaps just dropping `ignoring row`?

Reply:
I think we can put the line number in the error message as well. I'll update with that too.

Reply:
I predict: 1. ^ is a good local improvement but 2. folks are going to want it changed again, and again, to report the .. line number, the line itself, stuff like that. The PR does a good job linking up errors with the output, and allows folks to iterate on the error content.